src/Compilers/VisualBasic/Portable/Symbols/Metadata/PE/PEMethodSymbol.vb

```diff
@@ -1136,12 +1150,25 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols.Metadata.PE
             If Not _packedFlags.IsUseSiteDiagnosticPopulated Then
                 Dim errorInfo As DiagnosticInfo = CalculateUseSiteErrorInfo()
                 EnsureTypeParametersAreLoaded(errorInfo)
+                CheckUnmanagedCallersOnly(errorInfo)
                 Return InitializeUseSiteErrorInfo(errorInfo)
             End If
 
             Return _uncommonFields?._lazyUseSiteErrorInfo
         End Function
 
+        Private Sub CheckUnmanagedCallersOnly(ByRef errorInfo As DiagnosticInfo)
+            If Not _packedFlags.IsUseSiteDiagnosticPopulated AndAlso (errorInfo Is Nothing OrElse errorInfo.Code <> ERRID.ERR_UnsupportedMethod1) Then
```

Comment:
>Not _packedFlags.IsUseSiteDiagnosticPopulated AndAlso [](start = 15, length = 53)

I don't think we need to check this. If you going after optimization, I would prefer us to not call this method rather than changing its outcome based on optimization.